pkg/secrets/pushtofile/secrets_groups.go

```diff
@@ -0,0 +1,220 @@
+package pushtofile
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/log/messages"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	CONJUR_SECRETS_PREFIX             = "conjur.org/conjur-secrets."
+	CONJUR_SECRETS_POLICY_PATH_PREFIX = "conjur.org/conjur-secrets-policy-path."
+	SECRET_FILE_PATH_PREFIX           = "conjur.org/secret-file-path."
+	SECRET_FILE_FORMAT_PREFIX         = "conjur.org/secret-file-format."
+	SECRET_FILE_TEMPLATE_PREFIX       = "conjur.org/secret-file-template."
+
+	DEFAULT_FILE_PERMS os.FileMode = 0777
+	DEFAULT_FILE_PATH  string      = "/conjur/secrets/"
+)
+
+type SecretsFileFormat int
+
+const (
+	FILE_FORMAT_NONE SecretsFileFormat = iota
+	FILE_FORMAT_YAML
+	FILE_FORMAT_JSON
+	FILE_FORMAT_DOTENV
+	FILE_FORMAT_BASH
+	FILE_FORMAT_PLAINTEXT
+)
+
+// SecretsPaths comprises Conjur variable paths for all secrets in a secrets
+// group, indexed by secret name.
+type SecretsPaths map[string]string
+
+// SecretsGroupInfo comprises secrets mapping information for a given secrets
+// group.
+type SecretsGroupInfo struct {
+	Secrets      SecretsPaths
+	FilePath     string
+	FileFormat   SecretsFileFormat
+	FilePerms    os.FileMode
+	FileTemplate string
+}
+
+// SecretsGroups comprises secrets mapping info for all secret groups
+type SecretsGroups map[string]SecretsGroupInfo
```

Comment:
So here the key of the map is the label? Looks good.

Reply:
Yep, the name of the secrets group

Reply:
Disagree but changed to avoid prolonged pointless bikeshedding

Reply:
I believe we're overthinking this. This is not a complex problem. Just use a map and add a parameter for the key to functions processing the group object